src/components/MyGraph.js

```diff
@@ -27,18 +32,29 @@ class MyGraph extends React.Component {
         const padding = 10;
         const graphHeight = this.props.size[1] - (axisBorder + padding);
 
-        const xScale = scaleLinear()
+        const xScale = scaleTime()
             .domain(extent(timestamps))
             .range([0, this.props.size[0] - (axisBorder + padding)]);
 
+
+        //Let's work out the widths of the bars here...
+        const widths = [];
+        for (var i = 0; i<this.props.data.length-2; i++) {
+            widths.push(1 + xScale(this.props.data[i+1][0]) - xScale(this.props.data[i][0]));
+        }
+        widths.push[1]; // special case on the Right hand end
```

Comment:
This is not right. But what should it be?

Reply:
Well, the brackets were wrong. Like how did that not have an error running. But the real problem was the boundary condition. I had -2 and stricty less than. Just need one of those.